### PR TITLE
Added NQueens in C++

### PR DESCRIPTION
**Pull request by YASH01009**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Description

(Added NQueens in cpp)

### Checklist

- [Y] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Issue #1524 is solved here)

### CODE REVIEW

File: C-Plus-Plus/cp/NQueens.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+#define N 20
+
+/**
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/NQueens.cpp:1]: Push the Algorithm Description above the Code

File: C-Plus-Plus/cp/NQueens.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            }
+    }
+}
+
+int main() {
+	int n;
+	cin >> n;
+
+	bool board[N][N];
+    bool cols[N];
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/NQueens.cpp:43]: Make the formatting consistent

File: C-Plus-Plus/cp/NQueens.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+**/
+    
+void solve(bool board[N][N], bool cols[N], bool ndiag[2*N-1], bool rdiag[2*N-1], int row, string asf, int n) {
+    
+    if(row == n) {
+        cout << asf << '.' << '\n';
+        return;
+    }
+    
+    for(int col=0; col<n; col++) {
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/NQueens.cpp:20]: Add some Comments explaining the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Code Looks good to me 👍

**Comment by YASH01009** [comment]
@HarshCasper changes done !!

**Comment by shraddhavp** [comment]
conflicts to be resolved @YASH01009

**Comment by HarshCasper** [comment]
@YASH01009 Please resolve the Conflicts

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**